src/zoid/buttons/prerender.jsx

```diff
@@ -31,8 +34,21 @@ export function PrerenderedButtons({ nonce, onRenderCheckout, props } : Prerende
             [ FPTI_KEY.CONTEXT_ID ]:         props.buttonSessionID,
             [ FPTI_KEY.TRANSITION ]:         'process_button_prerender_click'
         }).flush();
-        
-        if (supportsPopups()) {
+
+        if (canUseQRPay) {
+            const target : HTMLElement = event.currentTarget;
+            const spinner = target.querySelector(`.${ CLASS.SPINNER }`);
```

Comment:
What is the ideal behavior for the prerender click for QR Pay? My initial thought is it should launch a modal dialog with a loading spinner and second render replaces the content with the QR code. 

If we want to stick with this button loading spinner approach then can we make a helper function in button utils for adding this spinner? Something like `addButtonSpinner(element)`. That way we don't have class specific details in pre-render.

Reply:
You're right in wanting to throw up the modal right away, but I think that could potentially create a situation where we launch the modal but then we do a native app switch. Putting the venmo button in a "wait & see" mode and then calling `onRenderCheckout` and let the rendered components figure out how to handle that button click makes the most sense to me, otherwise I feel like we'd end up duplicating the quasi-eligibility "context checking" code that exists in for this in SPB. 

That being said, moving the functionality to add the spinner out of this file makes sense. I've moved it into `zoid/buttons/utils.js => showButtonLoading()`

Reply:
> You're right in wanting to throw up the modal right away, but I think that could potentially create a situation where we launch the modal but then we do a native app switch.

Good point. I didn't think about this use case.

> That being said, moving the functionality to add the spinner out of this file makes sense. I've moved it into zoid/buttons/utils.js => showButtonLoading()

Sounds good. Let's go with that